apps/main-site/src/pages/api/sitemap-checker.ts

```diff
@@ -0,0 +1,58 @@
+/* eslint-disable @typescript-eslint/restrict-template-expressions */
+/* eslint-disable @typescript-eslint/no-unsafe-argument */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import { NextApiRequest, NextApiResponse } from 'next';
+import { parseStringPromise } from 'xml2js';
+export default function handler(req: NextApiRequest, res: NextApiResponse) {
+	void fetchSitemap();
+}
+
+async function fetchSitemap() {
+	// fetching answeroverflow sitemap
+	const startTime = new Date();
+	const sitemapResponse = await fetch('https://Answeroverflow.com/sitemap.xml');
+	const endTime = new Date();
+	console.log(
+		`Time taken to fetch https://Answeroverflow.com/sitemap.xml: ${
+			endTime.getTime() - startTime.getTime()
+		}ms`,
+	);
+
+	// parsing answeroverflow sitemap's xml to js
+	const sitemapText = await sitemapResponse.text();
+	const data = await parseStringPromise(sitemapText);
+
+	// Fetching and parsing each sitemap then retrieving all urls from each sitemap
+	let sitemapCount = 0;
+	let urlCount = 0;
+	const startTime2 = new Date();
+	console.log('Fetching and parsing each sitemap...');
+	const promises = data.sitemapindex.sitemap.map(async (sitemap) => {
```

Comment:
Just again this needs to be typed, it's currently an `any` type.

Reply:
alright!